### ISSUE TITLE
Feature request: ability to format code files hosted on github

### ISSUE THREAD

**Issue by fxnoob** [prettier/prettier-chrome-extension#108]
When I am on GitHub in some project directory .if that's a file say index.js then by clicking on some extensions option I should get that file content rerendered with prettier formatted code

**Comment by nickmccurdy** [comment]
To be clear, do you mean a read-only display or within the file editor?

**Comment by fxnoob** [comment]
@nickmccurdy
In file editor, is ok to me. nice to have both.

**Comment by nickmccurdy** [comment]
The former should we easy as we would just need to tweak how we detect text fields and make sure the button styles are appropriate in this toolbar. I'm not sure if we can reliably do the latter, as even if we supported read-only formatting, it could interfere with GitHub features like highlighting, selections, line/range links, and comments.

**Comment by fxnoob** [comment]
> if we supported read-only formatting, it could interfere with GitHub features like highlighting, selections, line/range links, and comments.

Good point.
@nickmccurdy 
Maybe provide an option in content menu of chrome to get it formatted in the option page of this chrome extension

**Comment by nickmccurdy** [comment]
We're tracking that in #21

**Comment by kaicataldo** [comment]
Agreed with @nickmccurdy. I don’t think it’s a good idea to format read-only content. The overhead of reverse-engineering the DOM structure to work with GitHub’s scripts would be extremely difficult, with no guarantees that it’ll work at any given time.

**Comment by nickmccurdy** [comment]
I'm okay with keeping this open to track support for GitHub's file editors.

**Comment by hyperupcall** [comment]
I did some investigating and this seems less straightfoward. The file editor is CodeMirror 6 (with gist.github.com, it is CodeMirror 5). From what I know about CodeMirror 6, it tracks [state changes](https://codemirror.net/docs/ref/#state) internally (not reading DOM), so a solution may require writing an extension. But I doubt GitHub has some mechanism for customizing CodeMirror editor extensions.

**Comment by nickserv** [comment]
@hyperupcall I think that's just for code editors, not discussion editors, right? We could shift focus to supporting discussion editors since GitHub's VSCode based editors (replacing code editors) already support Prettier.
